src/js/components/Nav/Nav.tsx

```diff
@@ -0,0 +1,132 @@
+ *  This component offers a collapsable width.
+ *
+ * Please see the Storybook example for how we recommend using the Nav component.
+ *
+ */
+
+import * as React from "react";
+import * as PropTypes from "prop-types";
+import { Icon } from "../Icon";
+import { getClassName } from "../../utils/helpers";
+import { Avatar } from "../Avatar";
+
+const navPropTypes = {
+  // React component children
+  children: PropTypes.node,
+  // Custom classnames to add
+  className: PropTypes.string,
+  // TeamSnap UI modifiers to add to class name
+  mods: PropTypes.string,
+  // Custom React Styles to add to the nav
+  style: PropTypes.object,
+  // Any extraneous props
+  otherProps: PropTypes.object,
+  // a list of items to be displayed in the header component
+  headerItems: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      image: PropTypes.string,
+    })
+  ),
+};
+
+const itemPropTypes = {
+  // String that matches a teamsnap-ui icon name
+  icon: PropTypes.string,
+  // Teamsnap UI modifiers passed to the icon component
+  iconModifiers: PropTypes.string,
+  // bool wether this item should be marked as active or not, multiple can be marked.
+  isActive: PropTypes.bool,
+  // function to handle an item being clicked.
+  onClick: PropTypes.func,
+  // *external* link, creates an anchor tag under the hood. Use onClick and router state for SPA transitions
+  link: PropTypes.string,
+  // any props to pass to the generated anchor tag (target, for example)
+  linkProps: PropTypes.any
+};
+
+type NavType = React.FunctionComponent<
+  PropTypes.InferProps<typeof navPropTypes>
+>;
+type ItemType = React.FunctionComponent<
+  PropTypes.InferProps<typeof itemPropTypes>
+>;
+
+const Item: ItemType = ({ children, icon, iconModifiers, isActive, onClick, link, linkProps }) => {
+  const maybeIcon = icon ? <Icon name={icon} mods={`u-spaceRightXs ${iconModifiers}`} /> : null;
+  const Wrapper = link ? ({children}) => <a href={link} {...linkProps}>{children}</a> : ({children}) => <>{children}</>
+  return (
+    <Wrapper>
+      <li className={`${isActive ? `is-active ` : ``}nav-item`} onClick={onClick || (() => {})}>
+        {maybeIcon} <span className="nav-item-title">{children}</span>
+      </li>
+    </Wrapper>
+  );
+};
+
+const Nav: NavType & { Item: ItemType } = ({
+  className,
+  mods,
+  children,
+  style,
+  otherProps,
+  headerItems,
+}) => {
+  const [isCollapsed, setCollapsed] = React.useState(false);
+  const cname = getClassName(
+    "Nav",
+    isCollapsed && "is-collapsed",
+    className,
+    mods
+  );
+
+  return (
+    <nav className={cname} style={style} {...otherProps}>
+      {headerItems && headerItems.length ? (
+        <div className="nav-header u-textSemiBold u-flex">
+          <div
+            className={`${
+              !isCollapsed ? "u-size1of8" : ""
+            } nav-header-icon u-flex u-flexAlignItemsCenter u-flexJustifyCenter`}
```

Comment:
I don't think you need the `u-flex u-flexAlignItemsCenter u-flexJustifyCenter` on every child of the parent `nav-header u-flex` container. Adding `u-flexAlignItemsCenter` to the parent (nav-header) should do the trick.

However... if the nav header will always display flex, may be cleaner to move those to styles on the component:
```
.Nav-header {
  display: flex;
  align-items: center;
}
```

Reply:
I went ahead and pulled that out to Nav-header. It feels a lot cleaner, I think I'm just getting caught up trying to balance adding additionally CSS vs using existing classes. Ideally we can use this opportunity to shrink the CSS file, too, but I might need some guidance on that (or some help planning that out).

Reply:
The utility classes are useful to modify an existing component for one-off behavior changes, or for making layouts without having to create a whole new component for that one layout.

I don't typically think of them as a path to shrinking component CSS. Ideally component styles are completely encapsulated within the component classes, so that the component doesn't need any additional classes to display correctly. Our React components nullify that need a tad, but it's still best practice for CSS composition, and for if these components are ever used outside of React.

Reply:
The utility styles also compress very tightly due to their repetition. But I think we can probably sort through all of them to make sure we need/want them all.